### ISSUE TITLE
[Bug] Saving Patrol Craft CP page with authorised IP array

### ISSUE THREAD

**Issue by roberttolton** [selvinortiz/craft.patrol#21]
Hi, just saw this today pop up today.

Saving the Patrol configuration page in the Craft CP, when there is an array of authorised IP addresses set in `general.php` causes an error:

```
trim() expects parameter 1 to be string, array given

/vagrant/craft/plugins/patrol/services/PatrolService.php(253)

241         return $this->dynamicParams;
242     }
243 
244     /**
245      * Parses authorizedIps to ensure they are valid even when created from a string
246      *
247      * @param array|string $ips
248      *
249      * @return array
250      */
251     public function parseAuthorizedIps($ips)
252     {
253         $ips = trim($ips);
254 
255         if (is_string($ips) && ! empty($ips))
256         {
257             $ips = explode(PHP_EOL, $ips);
258         }
259 
260         return $this->filterOutArrayValues(
261             $ips,
262             function ($val)
263             {
264                 return preg_match('/^[0-9\.\*]{5,15}$/i', $val);
265             }
```

**Comment by selvinortiz** [comment]
@roberttolyon Is this on the latest release of Patrol? I remember coming across an issue like this one but can't remember if it was this particular one or something else.

**Comment by selvinortiz** [comment]
@roberttolton ^

**Comment by roberttolton** [comment]
@selvinortiz This is the latest Patrol, just updated to it today.

**Comment by selvinortiz** [comment]
@roberttolton Ok, I'll try to get a fix out later today or tomorrow.

**Comment by selvinortiz** [comment]
@roberttolton This should be fixed now via the newly cut release **1.4.2**. Let me know if you continue to have issues!
